presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -38,11 +40,12 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
 
 public final class SystemSessionProperties
 {
     public static final String OPTIMIZE_HASH_GENERATION = "optimize_hash_generation";
-    public static final String DISTRIBUTED_JOIN = "distributed_join";
+    public static final String JOIN_DISTRIBUTION_TYPE = "join_distribution_type";
```

Comment:
@findepi @rschlussel We can't just drop support for `distributed_join` session property as it will break existing queries.

Reply:
Maybe we could leave `distributed_join` as a "proxy" property:
* setting `distributed_join = true` would set `join_distribution_type = PARTITIONED`
* setting `distributed_join = false` would set `join_distribution_type = REPLICATED`
* not setting `distributed_join` would leave `join_distribution_type` as is (with the default being PARTITIONED)

(a bit thinking aloud, not sure how this would play with `show session` and the like)

Or, we can have two properties: deprecated `distributed_join` (unset by default) and new `join_distribution_type` (with the default being PARTITIONED).  `distributed_join` would take precedence when it is set.

Reply:
> deprecated distributed_join (unset by default) and new join_distribution_type (with the default being PARTITIONED). distributed_join would take precedence when it is set.

I like this more

Reply:
+1 for 

> Or, we can have two properties: deprecated distributed_join (unset by default) and new join_distribution_type (with the default being PARTITIONED). distributed_join would take precedence when it is set.

@dain @martint How do we usually go about deprecating properties?

Reply:
I checked with @martint before i started working on this, and he said 
> go for it. Removing/renaming config options and sessions properties is not a problem in general. We've done similar things in the past for other features

so it seems like there's not necessarily a consistent deprecation process vs. just removing and replacing.  But i agree that it would be a big effort to remove this all at once with no transition period (It's a very commonly used option, and people couldn't change their settings before the release, and then everything would break as soon as a new release was deployed).  So i agree deprecation is a better way to go here.

Reply:
Yes, we have do it carefully, especially if people are using it. But, ultimately, we should remove it. This is another instance where user-visible warnings would help.